library-udf/src/test/java/org/apache/iotdb/library/dmatch/DMatchTests.java

```diff
@@ -0,0 +1,782 @@
+              100, 0, 0, 0, 0, 6, 6, 6, 6));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d4(timestamp,s1,s2,s3,s4,s5,s6,s7,s8) values(%d,%d,%d,%d,%d,%d,%d,%d,%d))",
+              200, 2, 2, 2, 2, 7, 7, 7, 7));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d4(timestamp,s1,s2,s3,s4,s5,s6,s7,s8) values(%d,%d,%d,%d,%d,%d,%d,%d,%d))",
+              300, 3, 3, 3, 3, 0, 0, 0, 0));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d4(timestamp,s1,s2,s3,s4,s5,s6,s7,s8) values(%d,%d,%d,%d,%d,%d,%d,%d,%d))",
+              400, 4, 4, 4, 4, 9, 9, 9, 9));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d4(timestamp,s1,s2,s3,s4,s5,s6,s7,s8) values(%d,%d,%d,%d,%d,%d,%d,%d,%d))",
+              500, 5, 5, 5, 5, 10, 10, 10, 10));
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+  }
+
+  private static void registerUDF() {
+    try (Connection connection = EnvFactory.getEnv().getConnection();
+        Statement statement = connection.createStatement()) {
+      statement.execute(
+          "create function completeness as 'org.apache.iotdb.library.dmatch.UDAFCov'");
+      statement.execute("create function timeliness as 'org.apache.iotdb.library.dmatch.UDAFDtw'");
+      statement.execute(
+          "create function consistency as 'org.apache.iotdb.library.dmatch.UDAFPearson'");
+      statement.execute("create function validity as 'org.apache.iotdb.library.dmatch.UDTFPtnSym'");
+      statement.execute("create function validity as 'org.apache.iotdb.library.dmatch.UDTFXcorr'");
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+  }
+
+  @AfterClass
+  public static void tearDown() throws Exception {
+    EnvFactory.getEnv().cleanAfterClass();
+    ConfigFactory.getConfig()
+        .setUdfCollectorMemoryBudgetInMB(100)
+        .setUdfTransformerMemoryBudgetInMB(100)
+        .setUdfReaderMemoryBudgetInMB(100);
+  }
+
+  @Test
+  public void testCov1() {
+    String sqlStr = "select cov(d1.s1,d1.s5) from root.vehicle";
+    try (Connection connection = EnvFactory.getEnv().getConnection();
+        Statement statement = connection.createStatement()) {
+      ResultSet resultSet = statement.executeQuery(sqlStr);
+      String result1 = resultSet.getString(1);
+      Assert.assertEquals("12.291666666666666", result1);
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+  }
+
+  @Test
+  public void testCov2() {
+    String sqlStr = "select cov(d1.s2,d1.s6) from root.vehicle";
+    try (Connection connection = EnvFactory.getEnv().getConnection();
+        Statement statement = connection.createStatement()) {
+      ResultSet resultSet = statement.executeQuery(sqlStr);
+      String result1 = resultSet.getString(1);
```

Comment:
Missing `resultSet.next()` before this line?

Reply:
Done.